### ISSUE TITLE
IDnow inflating APK size

### ISSUE THREAD

**Issue by crysxd** [idnow/de.idnow.android#15]
We took a closer look on our APK size today and unfortunately discovered that IDnow makes up a way too large percentage of the APK size. For example: IDnow includes dozens of images with poor compression applied. I took the images stored in /drawable (12 JPEG files) and dropped them in a random WEBP converter. This operation saves 1.6 MB on the shipped APK (zips attached for comparision). As IDnow has also JPEGs in /drawable-hdpi and others, the possible impact is even greater. WEBP is supported by all android platforms from 14 on. I also realised that the images stored in /drawable have a way higher resolution (640 pixels wide) than the images stored in /drawble-xhdpi (200 pixels wide) but they seem to be used in the same context. Would it be possible to reduce the image resolution? This would drop the 2,6 MB to 250 Kb when using JPEG.

Overall, IDnow is inflating the download size of our app. Please reduce the impact the library has on APK size, small optimisations like properly compressing / resizing the images take minutes and will have a great effect.  

As a side node: I was able to greatly reduce the size by removing unnecessary .so files (we now only ship armeabi-v7a). Each dropped platform strips ~ 2MB from the APK size and there are not many devices which will not be supported any more (e.g x86 phones like the Asus ZenFone 2 which sold really badly).

[JPG.zip](https://github.com/idnow/de.idnow.android/files/1016474/JPG.zip)
[WEBP.zip](https://github.com/idnow/de.idnow.android/files/1016475/WEBP.zip)

**Comment by hkretzschmar** [comment]
Hi crysxc,

thanks for the time to investigate about reducing the size of the resources. I just created an internal ticket for this which will be looked into in the course of the upcoming SDK version 3.5.0.

Kind regards
Holger Kretzschmar

**Comment by BaranMichal25** [comment]
Why in version 4.0.0 there are still no SVGs?
especially considering that the minSdkVersion is 21

**Comment by mreichelt** [comment]
@crysxd good point! Also, I think there is a huge potential for image savings in the IDnow SDK. There are 17MB in GIFs alone:

![image](https://user-images.githubusercontent.com/400322/113866174-494f4400-97ad-11eb-858c-a577b2ec1563.png)

The largest GIF is 1.3MB. By converting it to an animated WebP (like the issue author suggests), this file becomes 184kb in size.

Also, by using Lottie from AirBnB, all these animations could be replaced by much smaller vector ones - adding only [287kb](https://airbnb.io/lottie/#/android?id=what-is-the-impact-of-lottie-on-apk-size) in uncompressed size for the Lottie runtime! That way, the SDK wouldn't even need to ship any variants for the different device densities, and could become way smaller.

**Comment by crysxd** [comment]
@mreichelt There were luckily no GIFs in 2017, they must have been added later. What I ended up doing is to extract all images, properly compress them and then add them to my app resources. Gradle will replace the large ones from the library even if the file extension is different (some glitches with vector compress in the last,

**Comment by AlonsoFloo** [comment]
@souheib93 Could you maybe have a look on this tickets ? Running an optimizer should take 2-3minutes of your time.

The Gif format is not optimized on compile time, can you at least optimized them yourself ?
I've use [ImageOptim](https://imageoptim.com/mac) to reduce from 1.3 to 767.

<img width="754" alt="Screenshot 2021-04-08 at 09 13 25" src="https://user-images.githubusercontent.com/3592969/113983861-b7961400-984a-11eb-9f4a-acae1c440df1.png">


**Not optimized**
![vid_beforestarting_dark_01](https://user-images.githubusercontent.com/3592969/113984002-e613ef00-984a-11eb-9d61-2decd961fd72.gif)
**Optimized**
![vid_beforestarting_dark_01_optimized](https://user-images.githubusercontent.com/3592969/113984006-e7451c00-984a-11eb-9b59-8ac44f472c3e.gif)

**Comment by mreichelt** [comment]
@AlonsoFloo good point - the easiest thing IDnow could do right now is to run an optimizer on all images / convert them to more efficient formats.

Although long-term, it would make sense to completely move to a vector-based animation format - like Lottie. That way the savings could be more than 10x instead of 2x. 🙂

**Comment by souheib93** [comment]
Hello,

Thank you for your insights.

Yes, it is already decided to optimise the images and replace all Gifs with Lottie files.

It will be released soon in one of the upcoming releases.

**Comment by mreichelt** [comment]
Thank you @souheib93 - that is wonderful news! 🎉

**Comment by AlonsoFloo** [comment]
@souheib93 since the changelogs do not explicit it. I manually checked the release 5.0.7.
All gifs have been migrated to Lottie files.

Is there any remaining stuff ? or can we do the update freely.

**Comment by AlonsoFloo** [comment]
Never mind. Found mutiple 1mb video file for animations.

<img width="501" alt="Screenshot 2021-06-04 at 18 00 29" src="https://user-images.githubusercontent.com/3592969/120830408-c1907680-c55e-11eb-89dc-99efa9a05dc4.png">

![ezgif-3-afec2d406646](https://user-images.githubusercontent.com/3592969/120830361-b3daf100-c55e-11eb-9563-55e15fdbb133.gif)


But it's getting better. Thanks @souheib93 🎉
